src/_path.h

```diff
@@ -887,9 +898,18 @@ bool path_intersects_path(PathIterator1 &p1, PathIterator2 &p2)
 
     c1.vertex(&x11, &y11);
     while (c1.vertex(&x12, &y12) != agg::path_cmd_stop) {
+        // if the segment in path 1 is 0 length, skip to next vertex
+        if ((x11 == x12) && (y11 == y12)) {
+            continue;
+        }
         c2.rewind(0);
         c2.vertex(&x21, &y21);
+
         while (c2.vertex(&x22, &y22) != agg::path_cmd_stop) {
+            // if the segment in path 2 is 0 length, skip to next vertex
+            if ((x21 == x22) && (y21 == y22)) {
+                continue;
```

Comment:
if you're really going to do the check to zero-len both here and in segments_intersect (which seems weird as well) at least the check should be the same in both places (len < epsilon)?

Reply:
My logic was this:

 - computing `==` is cheaper
 - if trip the `==` condition, we will trigger the determinant condition, but the converse is not true.
 - do the cheaper, stricter thing here
 - do the more expensive looser thing where we have to (see my discussion with @timhoffm above)
 - keep all of the "isclose" logic together in the code so we are sure that the numbers don't get out of sync

Reply:
Well you could change the signature of isclose() to just isclose(a, b) and force atol and rtol as constants in the body of the function...  I'm not going to block over that though.